io/disk/fiotest.py

```diff
@@ -138,7 +138,13 @@ def setUp(self):
                     if line.startswith(eng) and 'no' in line:
                         self.cancel("PMEM engines not built with fio")
 
-        if not self.disk:
+        if self.disk:
+            if self.disk not in disk.get_disks():
+                self.cancel("Missing disk %s in OS" % self.disk)
+        else:
+            self.cancel("Please Provide valid disk name")
+
+        if not self.dir:
```

Comment:
@AYUSHJAIN951 tried to understand the different yaml files we have for this test case and discuss with the end user of `fio-pmem.yaml,` they want to run the test on pmem chunk and may not be passing the disk all the time.

even I could see there is no disk parameter in `fio-perf.yaml` and they have hard coded the /mnt in the this yaml which indicates they will run this test on /mnt and without disk.

So if we check only disk here and cancel the test its not going to help other users.
so lets have a common yaml parameter across this yaml files.
parameter can be like, 

`run_on_disk : True`

So for test to run on any san/sas or on any raw disk we set it as `True` else `False`

Add above line as comment at this parameter in yaml.
So for fio.yaml, it will be `True` and for other yamls it will be `False`. hard code this in yaml accordingly.

Now in .py
```
if run_on_disk:
    if not self.disk:
         self.cancel ("provide the disk to run test")
if  not self.dir:
     self.dir = self.workdir
```

@narasimhan-v let me know your thoughts too on this to handle

Reply:
one more thing,
creating lvs/raids will be done only when the test is intended to run raw disk. and lvs and raids not applicable in other case.
So have this check in such a way that creation of lvs/raids will happen only when disk is provided and in other case it should not be ran.

let me know if you need any help here.

Reply:
@Naresh-ibm are we unnecessarily complicating the test case to handle all users of it ? I can see that every change is required to handle all types of inputs.
Can we have a separate test case for pmem and simplify the test rather than adding parameters which is confusing to anyone new to contribute ?

Reply:
Right now I can see the test case is having numerous parameters manipulated, like self.dir, self.workdir, etc. This is not intuitive and makes it difficult for anyone to send a quick fix. I strongly think we should just clone the test and have it separately for pmem and perf, since I do not see any contributions from those users for a long time now.

Reply:
How does this sound have YAML file part of upstream and in clone it will work, new YAML aligned with the user's needs, then we should be having a generic solution. and avoid to   change py file,